pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobDetails.tsx

```diff
@@ -59,6 +68,260 @@ enum TabKeysEnum {
   PROFILER = "Advanced Debugging",
 }
 
+export interface JobDetailsPropsV2<AppState = unknown>
+  extends RouteComponentProps {
+  refreshUserSQLRoles: () => void;
+  onFetchExecutionDetailFiles: (
+    req: ListJobProfilerExecutionDetailsRequest,
+  ) => Promise<ListJobProfilerExecutionDetailsResponse>;
+  onCollectExecutionDetails: (
+    req: CollectExecutionDetailsRequest,
+  ) => Promise<CollectExecutionDetailsResponse>;
+  onDownloadExecutionFile: (
+    req: GetJobProfilerExecutionDetailRequest,
+  ) => Promise<GetJobProfilerExecutionDetailResponse>;
+  adminRoleSelector: Selector<AppState, UIConfigState["hasAdminRole"]>;
+}
+
+export function JobDetailsV2<AppState = unknown>({
```

Comment:
Are we keeping this in the same file to preserve the file history?
That's great if so but I might still suggest splitting `OverviewTabContent` out if so.

Reply:
> Are we keeping this in the same file to preserve the file history?

Yeah that was the intention!

And yeah, I'll split it out, definitely looks cleaner that way.